SPIKE - db integration tests for dickon branch

driessamyn opened the pull request.

dickon committed:
use new schema

red bar: get TestV1CryptoRepositryImpl working with new schema

 update to latest schema

 update to latest schema

match on alias

refactor: use with more

untested code to allow multiple root keys

use new config schema and wrapping key schema
Does not work yet due to signing key differences being missing, as expected

use signing key entity

parallel builds

fix tests

fixes

get first persistence test code running

For now keyMaterial is nullable due to a weird Hibernate behaviour

drop eager annotation

fix test

port tests

simplify

port tests

do not reuse a wrapping key name

port test
update timestamp name

fixes

use artifactory API version

revert parallel builds

Commit:
Crypto repo DB tests.

dickon commented on components/crypto/crypto-softhsm-impl/src/test/kotlin/net/corda/crypto/softhsm/impl/infra/TestWrappingRepository.kt:
```diff
@@ -9,8 +9,9 @@ class TestWrappingRepository(
     val findCounter = mutableMapOf<String, Int>()
 
-    override fun saveKey(alias: String, key: WrappingKeyInfo) {
+    override fun saveKey(alias: String, key: WrappingKeyInfo): WrappingKeyInfo {
```
Curious what the reason for this is? I had considered doing the same, and it can help the calling code be concice (i.e. gets closer to the fluent style)

dickon commented on libs/crypto/crypto-config-impl/src/main/kotlin/net/corda/crypto/config/impl/CryptoConfigUtils.kt:
```diff
@@ -271,6 +272,10 @@ fun createDefaultCryptoConfig(wrappingKeyPassphrase: Any, wrappingKeySalt: Any):
                 )
             )
         )
+        .withValue("defaultWrappingKey", ConfigValueFactory.fromAnyRef("root1"))
```
this will go away when you merge/rebase my work today.

dickon commented on components/crypto/crypto-softhsm-impl/src/main/kotlin/net/corda/crypto/softhsm/impl/WrappingRepositoryImpl.kt:
```diff
@@ -10,28 +10,32 @@ import java.util.UUID
+    override fun saveKey(alias: String, key: WrappingKeyInfo): WrappingKeyInfo {
+        entityManagerFactory.createEntityManager().use { em ->
+            em.transaction {
+                return it.merge(
```
Is the difference between `merge` and `persist` around the unique constraint on the secondary keys?

Commit:
Fix up after rebase on Dickon's branch.

Commit:
Fix up unit tests and detekt
NOTE:
- SigningRepositoryTest: commented out all tests, I don't think it's worth having these as Unit tests
- TestInitialConfigPluginCrypto: tests disabled for now
